### ISSUE TITLE
Handle suppressed errors - Suggestion

### ISSUE THREAD

**Issue by wehowski** [frdl/oidplus-frdlweb-rdap#30]
Hello Daniel,
if you use '@' e.g. at 
````
@file_put_contents($cacheFile, serialize($out));
````
https://github.com/frdl/oidplus-frdlweb-rdap/blob/main/plugins/frdl/publicPages/1276945_rdap/OIDplusRDAP.class.php#L233
and the file could not be written, e.g. because the cache directory does not exist, the user/admin will not recognize the error.

I suggest (if you need to suppress the "raw" error), you could handle it, e.g. like:
````
if(!@file_put_contents($cacheFile, serialize($out)) ){
   // log or trigger any (admin) error or warning...

}
````

**Comment by danielmarschall** [comment]
Indeed, I had this issue where the cache directory was write-protected, and I just noticed that the app was slow (due to missing caching). I do not want a hard-error that prevents the system from working, so I used `@` .

Logging the error has the problem that the log-file would be filled with thousands of errors (because each page load is 1 log entry).

I have an idea: Our new "notification" plugin could detect such errors ("cache-directory write-protected"). I just need to figure out how to do this, because the "notification" plugin asks other plugins for their notifications, and the cache is something the system manages. Maybe I need to create some kind of invisible plugin for that task...
